podcast.php

```diff
@@ -122,6 +156,48 @@ public static function retreiveAudioDuration($file)
     }
 
     /**
+     * Retrieve audio metadata filesize.
+     *
+     * @param string $file
+     *     Path to audio file.
+     * @return string
+     *     Audio filesize.
+     */
+    public static function retreiveAudioLength($file)
+    {
+        $id3 = GetID3Plugin::analyzeFile($file);
+        return ($id3['filesize']);
+    }
+ 
+    /**
+     * Retrieve remote audio file to /tmp directory
+     *
+     * @param string $url
+     *     url audio file.
+     * @return string
+     *     Audio filesize.
+     */
+    public static function getRemoteAudio($url)
+    {
+        if ($fp_remote = fopen($url, 'rb')) 
+        {
+            $localtempfilename = tempnam('/tmp', 'getID3');
+            if ($fp_local = fopen($localtempfilename, 'wb'))
+            {
+                while ($buffer = fread($fp_remote, 8192)) 
```

Comment:
Example 3 on this page: http://php.net/manual/en/function.fread.php
has some better remote file handling.

Reply:
agreed, I grabbed this one from the getid3 php website - I should have looked a little closer at it. I can redo this for sure and make it better base don that exampel.